src/main/java/org/testng/TestNG.java

```diff
@@ -269,28 +269,33 @@ public void setXmlPathInJar(String xmlPathInJar) {
   public void initializeSuitesAndJarFile() {
     // The Eclipse plug-in (RemoteTestNG) might have invoked this method already
     // so don't initialize suites twice.
+	System.out.println("ENTER");
     if (m_isInitialized) {
       return;
     }
 
     m_isInitialized = true;
     if (m_suites.size() > 0) {
-    	//to parse the suite files (<suite-file>), if any
-    	for (XmlSuite s: m_suites) {
-        for (String suiteFile : s.getSuiteFiles()) {
-            Path rootPath = Paths.get(s.getFileName()).getParent();
-            try {
-                Collection<XmlSuite> childSuites = getParser(rootPath.resolve(suiteFile).normalize().toString()).parse();
-                for (XmlSuite cSuite : childSuites){
-                    cSuite.setParentSuite(s);
-                    s.getChildSuites().add(cSuite);
-                }
+    //to parse the suite files (<suite-file>), if any
+	for (XmlSuite s: m_suites) {
+	   for (String suiteFile : s.getSuiteFiles()) {
+        	Path suitePath = Paths.get(s.getFileName());
```

Comment:
s.getFileName() can return null in surefire plugin suites. By putting it here we will avoid the surefire plugin case and avoid a null check.